### PR TITLE
feat: add boolean type for builtins.html[0].inject

### PR DESCRIPTION
**Pull request by lippzhang**
## Summary

add boolean type for builtins.html[0].inject

https://github.com/web-infra-dev/rspack/issues/3761

## Test Plan

I wrote unit tests inside the test suite

### CODE REVIEW

File: crates/rspack_plugin_html/src/config.rs [hunk 1]
```diff
@@ -9,11 +9,13 @@ use sugar_path::SugarPath;
 use crate::sri::HtmlSriHashFunction;
 
 #[cfg_attr(feature = "testing", derive(JsonSchema))]
-#[derive(Deserialize, Debug, Clone, Copy)]
+#[derive(Deserialize, Debug, Clone, Copy, PartialEq)]
```
Comment (hyf0) [inline crates/rspack_plugin_html/src/config.rs:12]: Why add `PartialEq`?

File: crates/rspack_plugin_html/src/config.rs [hunk 2]
```diff
@@ -107,7 +116,7 @@ impl Default for HtmlPluginConfig {
       template: None,
       template_content: None,
       template_parameters: None,
-      inject: None,
+      inject: default_inject(),
```
Comment (hyf0) [inline crates/rspack_plugin_html/src/config.rs:119]: Why not keep `None`?

File: crates/rspack_plugin_html/src/config.rs (hunk 1 above)
Comment (lippzhang) [reply crates/rspack_plugin_html/src/config.rs:12]: When inject is 'true' or 'false', conditional operators need to be used for judgment processing.

Comment (hyf0) [reply crates/rspack_plugin_html/src/config.rs:12]: where is the related code? Could you point it to me?

File: crates/rspack_plugin_html/src/config.rs (hunk 2 above)
Comment (lippzhang) [reply crates/rspack_plugin_html/src/config.rs:119]: I read the source code of [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin),  the default value of inject is 'true'.  I'll change it.  
Doing this, you can reduce `create_script` and `create_style` function。

the typescript side handles the case of `undefined`  in  [packages/rspack/src/config/builtins.ts](https://github.com/web-infra-dev/rspack/pull/3771/files#diff-908957625f9952a51b579835fc42cf2a8d717ac0ffa1ce16aa4ac24bf6362885)

File: crates/rspack_binding_options/src/options/raw_builtins/raw_html.rs [hunk 3]
```diff
@@ -47,16 +47,24 @@ pub struct RawHtmlPluginConfig {
...
+        .expect("value.inject has unwrap_or_else so this will never happen");
 
     let script_loading = HtmlPluginConfigScriptLoading::from_str(
       &value
         .script_loading
         .unwrap_or_else(|| String::from("defer")),
     )
     .expect("value.script_loading has unwrap_or_else so this will never happen");
+    // Passing `true` will add it to the head/body depending on the `scriptLoading` option. Reference: https://github.com/jantimon/html-webpack-plugin
+    if inject == HtmlPluginConfigInject::True {
```
Comment (lippzhang) [inline crates/rspack_binding_options/src/options/raw_builtins/raw_html.rs:61]: the code is here  for PartialEq  @hyf0

File: crates/rspack_plugin_html/src/config.rs (hunk 1 above)
Comment (lippzhang) [reply crates/rspack_plugin_html/src/config.rs:12]: > where is the related code? Could you point it to me?

ok

File: crates/rspack_binding_options/src/options/raw_builtins/raw_html.rs (hunk 3 above)
Comment (hyf0) [reply crates/rspack_binding_options/src/options/raw_builtins/raw_html.rs:61]: Using `matches!(inject, HtmlPluginConfigInject::True)` would be enough.

File: packages/rspack/src/config/zod/builtins.ts [hunk 4]
```diff
@@ -30,8 +30,8 @@ export function builtins() {
 				filename: z.string().optional(),
 				template: z.string().optional(),
 				templateParameters: z.record(z.string()).optional(),
-				inject: z.enum(["head", "body"]).optional(),
-				publicPath: z.string().optional(),
+				inject: z.enum(["head", "body"]).or(z.boolean()).optional(),
+				publicPathz: z.string().optional(),
```
Comment (hyf0) [inline packages/rspack/src/config/zod/builtins.ts:34]: Typo.

**Review by hyf0 — DISMISSED** [review]
LGTM

**Comment by ScriptedAlchemy** [comment]
is the reason we cannot merge this due to windows CI failure?

**Comment by hyf0** [comment]
> is the reason we cannot merge this due to windows CI failure?

I haven't dig into this, but probably.

**Review by ahabhgk — APPROVED** [review]

**Pull request merged**